server/app/services/CfJsonDocumentContext.java

```diff
@@ -89,6 +91,21 @@ public boolean hasValueAtPath(Path path) {
     }
   }
 
+  public void putPhoneNumber(Path path, String phoneNumber) {
+    if (phoneNumber.isEmpty()) {
+      putNull(path);
+      return;
+    }
+    // Currently on US and CA numbers are supported, hence testing for just the two regions.
+    // for future cases, it is better to get the countrycode and test against the country code, not
+    // just all possible regions.
+    if (PHONE_NUMBER_UTIL.isPossibleNumber(phoneNumber, "US")
+        || PHONE_NUMBER_UTIL.isPossibleNumber(phoneNumber, "CA")) {
+      put(path, phoneNumber.replaceAll("[^0-9]", ""));
+    } else {
+      throw new IllegalArgumentException("invalid_phone_input");
```

Comment:
**suggestion (blocking):** include more info in the exception e.g. "Invalid phone number format", consider also including the format in the exception with the numbers redacted e.g. `String.format("Invalid phone number format: %s", phoneNumber.replaceAll("\\d", "X"))`

Reply:
Thanks Bion. Actually this not only catches formatting error but also invalid number like 555-###-####, though they look legitimate, they are not actual phone numbers. I have updated the comment in this method.